src/openms/source/CHEMISTRY/ElementDB.cpp

```diff
@@ -574,6 +570,11 @@ namespace OpenMS
 
   void ElementDB::addElementToMaps_(const string& name, const string& symbol, const unsigned int an, const Element* e)
   {
+    // delete existing element before adding a new one
+    if (atomic_numbers_.find(an) != atomic_numbers_.end())
+    {
+      delete atomic_numbers_[an];
```

Comment:
there is a major issue here: if we delete the Element ptr, then any structure that uses that ptr has a problem, eg any `EmpiricalFormula` that already exists will run into a segfault...

Reply:
ok so we could at least in principle modify it here (which will not update anything that was calculated in a lazy way though) - and did I mention that I hate those singletons ;)

Reply:
yes, I think we could try 

```
*atomic_numbers_[an] = *e;
```